lib/cgus.py

```diff
@@ -0,0 +1,62 @@
+import json
+import logging
+from pathlib import Path, PosixPath
+
+import nltk
+
+logging.basicConfig(level=logging.INFO)
+
+
+class CGUsDataset():
+    """
+    Helper class for handling CGUs Versions
+    """
+
+    def __init__(self, root_path: str = "../CGUs-versions"):
+        self.root_path = Path(root_path)
+        assert self.root_path.exists(), f"{root_path} does not exist"
+        assert self.root_path.is_dir(), f"{root_path} is not a directory"
+
+
+    def yield_all_md(self, ignore_rootdir: bool = True) -> list:
+        if ignore_rootdir:
+            return self.root_path.glob('**/*/*.md')
+        else:
+            return self.root_path.glob('**/*.md')
+
+class CGU():
+    """
+    A CGU object, and the main building bloc for our algorithms.
+    """
+
+    # Remove all punctuation when tokenizing text
+    _TOKENIZER = nltk.tokenize.RegexpTokenizer(r'\w+')
+
+    def __init__(self, path: PosixPath):
+        self._path = path
+        self.name = self._path.name
+        self.source = self._path.as_posix().split("/")[-2]
+        self.fullname = f"{self.source} - {self.name}".removesuffix(".md")
+        self.cgu_type = f"{self.name}".removesuffix(".md") #TODO check that this is the way to go about it
+        self.raw_content = self._path.read_text()
+        self.tokens = self._TOKENIZER.tokenize(self.raw_content)
+
+
+    def to_dict(self) -> dict:
+        """
+            "Serialize" the CGU object to key/value pairs.
+        """
+        return {
+            "cgu_type": self.cgu_type,
+            "num_words": len(self)
+        }
+
+    def most_used_words(self, n: int = 10):
```

Comment:
It seems it's not used for now?

Reply:
not yet as the results I got weren't too interesting imo
i'll remove this method if I end up not using it tomorrow